xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -414,3 +494,946 @@ func (s) TestEDSResourceRemoved(t *testing.T) {
+		},
+		Endpoints: []*v3endpointpb.ClusterLoadAssignment{
+			e2e.DefaultEndpoint(clusterName1, "localhost", []uint32{uint32(ports[0])}),
+			e2e.DefaultEndpoint(clusterName2, "localhost", []uint32{uint32(ports[1])}),
+		},
+		SkipValidation: true,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an xDS xdsClient for use by the cluster_resolver LB policy.
+	xdsClient, close, err := xdsclient.NewWithBootstrapContentsForTesting(bootstrapContents)
+	if err != nil {
+		t.Fatalf("Failed to create xDS client: %v", err)
+	}
+	defer close()
+
+	// Create a manual resolver and push a service config specifying the use of
+	// the cds LB policy as the top-level LB policy, and a corresponding config
+	// with a single cluster.
+	r := manual.NewBuilderWithScheme("whatever")
+	jsonSC := fmt.Sprintf(`{
+			"loadBalancingConfig":[{
+				"cds_experimental":{
+					"cluster": "%s"
+				}
+			}]
+		}`, clusterName)
+	scpr := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(jsonSC)
+	r.InitialState(xdsclient.SetClient(resolver.State{ServiceConfig: scpr}, xdsClient))
+
+	// Create a ClientConn and make a successful RPC.
+	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
+	if err != nil {
+		t.Fatalf("Failed to dial local test server: %v", err)
+	}
+	defer cc.Close()
+
+	// Make an RPC and ensure that it gets routed to cluster-1 since it is
+	// implicitly higher priority than cluster-2.
+	client := testgrpc.NewTestServiceClient(cc)
+	peer := &peer.Peer{}
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(peer), grpc.WaitForReady(true)); err != nil {
+		t.Fatalf("EmptyCall() RPC failed: %v", err)
+	}
+	if peer.Addr.String() != addrs[0].Addr {
+		t.Fatalf("EmptyCall() RPC routed to backend %q, want %q", peer.Addr, addrs[0].Addr)
+	}
+
+	// Swap the priorities of the EDS clusters in the aggregate cluster.
+	resources.Clusters = []*v3clusterpb.Cluster{
+		makeAggregateClusterResource(clusterName, []string{clusterName2, clusterName1}),
+		e2e.DefaultCluster(clusterName1, "", e2e.SecurityLevelNone),
+		e2e.DefaultCluster(clusterName2, "", e2e.SecurityLevelNone),
+	}
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Make an RPC and ensure that it gets routed to cluster-1 since it is
+	// implicitly higher priority than cluster-2.
+	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
+		if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(peer), grpc.WaitForReady(true)); err != nil {
+			t.Fatalf("EmptyCall() RPC failed: %v", err)
+		}
+		if peer.Addr.String() != addrs[1].Addr {
+			break
+		}
+	}
+	if ctx.Err() != nil {
+		t.Fatal("Timeout waiting for RPCs to be routed to cluster-2 after priority switch")
+	}
```

Comment:
Whether or not it needs to be routed to cluster-2 is orthogonal to this comment. This comment is rather than repeating the setup steps in the earlier test, to just move this assertion (and the switching of priorities) to the test above with a comment, to get rid of the duplicate code. I don't feel strongly either way.